### DISCUSSION TITLE
How to pass a kwarg named `ref` to a Python function

### DISCUSSION THREAD

**Discussion by rectalogic** [modular/modular#3431] [Q&A] [answered]
`librosa` has an API [librosa.power_to_db()](https://librosa.org/doc/latest/generated/librosa.power_to_db.html#librosa-power-to-db) that requires a kwarg named `ref`. How do I pass that from mojo, it seems to interpret `ref` as the reference keyword or something.

```mojo
from python import Python

fn main() raises:
    var librosa = Python.import_module("librosa")
    var np = Python.import_module("numpy")
    var y = librosa.load(librosa.ex('trumpet'))[0]

    librosa.power_to_db(y, ref=np.max)
```
```sh-session
> mojo run p.mojo
/Users/aw/Projects/rectalogic/experiments/mojo/p.mojo:8:28: error: unexpected token in expression
    librosa.power_to_db(y, ref=np.max)
                           ^
mojo: error: failed to parse the provided Mojo source module
```
In Python I would pass a dict literal like `librosa.power_to_db(y, **{"ref": np.max})` but dict literals are not supported in mojo.

Constructing a dict and unpacking is also not supported:
```mojo
    var kwargs = Python.dict()
    kwargs["ref"] = np.max
    librosa.power_to_db(y, **kwargs)
```
```sh-session
/Users/aw/Projects/rectalogic/experiments/mojo/p2.mojo:10:28: error: keyword unpacking not supported yet
    librosa.power_to_db(y, **kwargs)
                           ^
mojo: error: failed to parse the provided Mojo source module
```

**Comment by soraros** [comment, marked as answer]
You can write 
```mojo
librosa.power_to_db(y, `ref`=np.max)
```

**Reply by rectalogic** [reply]
Thanks, that works. I found documentation describing that:
> to avoid migration errors with Python code that uses identifier names that match Mojo keywords, Mojo provides a backtick feature that allows any keyword to behave as an identifier.

https://docs.modular.com/mojo/why-mojo#intentional-differences-from-python
